python/ray/tune/tests/test_progress_reporter.py

```diff
@@ -67,3 +184,50 @@ def testAddMetricColumn(self):
         reporter = CLIReporter()
         reporter.add_metric_column("foo", "bar")
         self.assertIn("foo", reporter._metric_columns)
+
+    def testProgressStr(self):
+        trials = []
+        for i in range(5):
+            t = Mock()
+            if i == 0:
+                t.status = "TERMINATED"
+            elif i == 1:
+                t.status = "PENDING"
+            else:
+                t.status = "RUNNING"
+            t.trial_id = "%05d" % i
+            t.local_dir = "/foo"
+            t.location = "here"
+            t.config = {"a": i, "b": i * 2}
+            t.evaluated_params = t.config
+            t.last_result = {"config": {"a": i, "b": i * 2}}
+            t.__str__ = lambda self: self.trial_id
+            trials.append(t)
+        prog1 = trial_progress_str(trials, ["a", "b"], fmt="psql", max_rows=3)
+        print(prog1)
+        assert prog1 == EXPECTED_RESULT_1
+        prog2 = trial_progress_str(
+            trials, ["a", "b"], fmt="psql", max_rows=None)
+        print(prog2)
+        assert prog2 == EXPECTED_RESULT_2
+
+    def testEndToEndReporting(self):
+        with tempfile.NamedTemporaryFile(suffix=".py") as f:
+            f.write(END_TO_END_COMMAND.encode("utf-8"))
+            f.flush()
+            output = subprocess.check_output(["python", f.name])
+            output = output.decode("utf-8")
```

Comment:
doesn't Ray print out random stuff here too? let me check if this test works on my machine

Reply:
I've managed to avoid the non-deterministic bits of Tune reporting I think. Ray does print some system info, but the checks below use "contains" instead of "equals" so those are ignored.